### ISSUE TITLE
GDT cannot be installed for Eclipse 2022-12 / 4.26 (package release)

### ISSUE THREAD

**Issue by flozano** [groovy/groovy-eclipse#1425]
(EDIT: same as #1399 latest comments, but that issue is closed)

I just downloaded a clean 2022-12 JEE package from Eclipse.org and installed it
(eclipse-jee-2022-12-R-macosx-cocoa-aarch64.dmg)

when I tried to add the GRECLIPSE snapshot 4.26 update site and install it, I got:
<img width="1125" alt="image" src="https://user-images.githubusercontent.com/1159291/207280040-bf43b2bd-ed7c-41e7-9bf1-9751b9adbb02.png">

```
Cannot complete the install because of a conflicting dependency.
  Software being installed: Eclipse Groovy Development Tools 4.8.0.v202212121946-e2212 (org.codehaus.groovy.eclipse.feature.feature.group 4.8.0.v202212121946-e2212)
  Software currently installed: Eclipse IDE for Enterprise Java and Web Developers 4.26.0.20221201-1200 (epp.package.jee 4.26.0.20221201-1200)
  Software currently installed: Eclipse Java Development Tools 3.18.1400.v20221123-1800 (org.eclipse.jdt.feature.group 3.18.1400.v20221123-1800)
  Only one of the following can be installed at once: 
    Java Development Tools Core 3.32.0.v202212121946-e2212 (org.eclipse.jdt.core 3.32.0.v202212121946-e2212)
    Java Development Tools Core 3.32.0.v20221108-1853 (org.eclipse.jdt.core 3.32.0.v20221108-1853)
    Java Development Tools Core 3.31.0.v20220831-1439 (org.eclipse.jdt.core 3.31.0.v20220831-1439)
  Cannot satisfy dependency:
    From: Eclipse IDE for Enterprise Java and Web Developers 4.26.0.20221201-1200 (epp.package.jee 4.26.0.20221201-1200)
    To: org.eclipse.equinox.p2.iu; org.eclipse.epp.package.common.feature.feature.group [4.26.0.20221201-1200,4.26.0.20221201-1200]
  Cannot satisfy dependency:
    From: Groovy-Eclipse Compilerless feature 4.8.0.v202212121946-e2212 (org.codehaus.groovy.compilerless.feature.feature.group 4.8.0.v202212121946-e2212)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.headless.feature.feature.group [4.8.0.v202212121946-e2212,4.8.0.v202212121946-e2212]
  Cannot satisfy dependency:
    From: Eclipse Groovy Development Tools 4.8.0.v202212121946-e2212 (org.codehaus.groovy.eclipse.feature.feature.group 4.8.0.v202212121946-e2212)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.compilerless.feature.feature.group [4.8.0.v202212121946-e2212,4.8.0.v202212121946-e2212]
  Cannot satisfy dependency:
    From: Groovy-Eclipse Headless feature 4.8.0.v202212121946-e2212 (org.codehaus.groovy.headless.feature.feature.group 4.8.0.v202212121946-e2212)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.jdt.patch.feature.group [4.8.0.v202212121946-e2212,4.8.0.v202212121946-e2212]
  Cannot satisfy dependency:
    From: Eclipse Packaging Project Common Feature 4.26.0.20221201-1200 (org.eclipse.epp.package.common.feature.feature.group 4.26.0.20221201-1200)
    To: org.eclipse.equinox.p2.iu; org.eclipse.jdt.core [3.32.0.v20221108-1853,3.32.0.v20221108-1853]
  Cannot satisfy dependency:
    From Patch: org.codehaus.groovy.jdt.patch.feature.group 4.8.0.v202212121946-e2212 Eclipse Java Development Tools 3.18.1400.v20221123-1800 (org.eclipse.jdt.feature.group 3.18.1400.v20221123-1800)
    To: org.eclipse.equinox.p2.iu; org.eclipse.jdt.core [3.32.0.v202212121946-e2212,3.32.0.v202212121946-e2212]
```

**Comment by flozano** [comment]
Same result when installing with a clean eclipse from the installer instead of a package:
```
Cannot complete the install because of a conflicting dependency.
  Software being installed: Eclipse Groovy Development Tools 4.8.0.v202212121946-e2212 (org.codehaus.groovy.eclipse.feature.feature.group 4.8.0.v202212121946-e2212)
  Software currently installed: Eclipse IDE for Enterprise Java and Web Developers 4.26.0.20221201-1200 (epp.package.jee 4.26.0.20221201-1200)
  Software currently installed: Eclipse Java Development Tools 3.18.1400.v20221123-1800 (org.eclipse.jdt.feature.group 3.18.1400.v20221123-1800)
  Only one of the following can be installed at once: 
    Java Development Tools Core 3.32.0.v202212121946-e2212 (org.eclipse.jdt.core 3.32.0.v202212121946-e2212)
    Java Development Tools Core 3.32.0.v20221108-1853 (org.eclipse.jdt.core 3.32.0.v20221108-1853)
  Cannot satisfy dependency:
    From: Eclipse IDE for Enterprise Java and Web Developers 4.26.0.20221201-1200 (epp.package.jee 4.26.0.20221201-1200)
    To: org.eclipse.equinox.p2.iu; org.eclipse.epp.package.common.feature.feature.group [4.26.0.20221201-1200,4.26.0.20221201-1200]
  Cannot satisfy dependency:
    From: Groovy-Eclipse Compilerless feature 4.8.0.v202212121946-e2212 (org.codehaus.groovy.compilerless.feature.feature.group 4.8.0.v202212121946-e2212)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.headless.feature.feature.group [4.8.0.v202212121946-e2212,4.8.0.v202212121946-e2212]
  Cannot satisfy dependency:
    From: Eclipse Groovy Development Tools 4.8.0.v202212121946-e2212 (org.codehaus.groovy.eclipse.feature.feature.group 4.8.0.v202212121946-e2212)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.compilerless.feature.feature.group [4.8.0.v202212121946-e2212,4.8.0.v202212121946-e2212]
  Cannot satisfy dependency:
    From: Groovy-Eclipse Headless feature 4.8.0.v202212121946-e2212 (org.codehaus.groovy.headless.feature.feature.group 4.8.0.v202212121946-e2212)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.jdt.patch.feature.group [4.8.0.v202212121946-e2212,4.8.0.v202212121946-e2212]
  Cannot satisfy dependency:
    From: Eclipse Packaging Project Common Feature 4.26.0.20221201-1200 (org.eclipse.epp.package.common.feature.feature.group 4.26.0.20221201-1200)
    To: org.eclipse.equinox.p2.iu; org.eclipse.jdt.core [3.32.0.v20221108-1853,3.32.0.v20221108-1853]
  Cannot satisfy dependency:
    From Patch: org.codehaus.groovy.jdt.patch.feature.group 4.8.0.v202212121946-e2212 Eclipse Java Development Tools 3.18.1400.v20221123-1800 (org.eclipse.jdt.feature.group 3.18.1400.v20221123-1800)
    To: org.eclipse.equinox.p2.iu; org.eclipse.jdt.core [3.32.0.v202212121946-e2212,3.32.0.v202212121946-e2212]
```

**Comment by flozano** [comment]
Same result with a clean 2022-12 installation from eclipse installer:
```
Cannot complete the install because of a conflicting dependency.
  Software being installed: Eclipse Groovy Development Tools 4.8.0.v202212121946-e2212 (org.codehaus.groovy.eclipse.feature.feature.group 4.8.0.v202212121946-e2212)
  Software currently installed: Eclipse IDE for Java Developers 4.26.0.20221201-1200 (epp.package.java 4.26.0.20221201-1200)
  Software currently installed: Eclipse Java Development Tools 3.18.1400.v20221123-1800 (org.eclipse.jdt.feature.group 3.18.1400.v20221123-1800)
  Only one of the following can be installed at once: 
    Java Development Tools Core 3.32.0.v202212121946-e2212 (org.eclipse.jdt.core 3.32.0.v202212121946-e2212)
    Java Development Tools Core 3.32.0.v20221108-1853 (org.eclipse.jdt.core 3.32.0.v20221108-1853)
  Cannot satisfy dependency:
    From: Eclipse IDE for Java Developers 4.26.0.20221201-1200 (epp.package.java 4.26.0.20221201-1200)
    To: org.eclipse.equinox.p2.iu; org.eclipse.epp.package.common.feature.feature.group [4.26.0.20221201-1200,4.26.0.20221201-1200]
  Cannot satisfy dependency:
    From: Groovy-Eclipse Compilerless feature 4.8.0.v202212121946-e2212 (org.codehaus.groovy.compilerless.feature.feature.group 4.8.0.v202212121946-e2212)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.headless.feature.feature.group [4.8.0.v202212121946-e2212,4.8.0.v202212121946-e2212]
  Cannot satisfy dependency:
    From: Eclipse Groovy Development Tools 4.8.0.v202212121946-e2212 (org.codehaus.groovy.eclipse.feature.feature.group 4.8.0.v202212121946-e2212)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.compilerless.feature.feature.group [4.8.0.v202212121946-e2212,4.8.0.v202212121946-e2212]
  Cannot satisfy dependency:
    From: Groovy-Eclipse Headless feature 4.8.0.v202212121946-e2212 (org.codehaus.groovy.headless.feature.feature.group 4.8.0.v202212121946-e2212)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.jdt.patch.feature.group [4.8.0.v202212121946-e2212,4.8.0.v202212121946-e2212]
  Cannot satisfy dependency:
    From: Eclipse Packaging Project Common Feature 4.26.0.20221201-1200 (org.eclipse.epp.package.common.feature.feature.group 4.26.0.20221201-1200)
    To: org.eclipse.equinox.p2.iu; org.eclipse.jdt.core [3.32.0.v20221108-1853,3.32.0.v20221108-1853]
  Cannot satisfy dependency:
    From Patch: org.codehaus.groovy.jdt.patch.feature.group 4.8.0.v202212121946-e2212 Eclipse Java Development Tools 3.18.1400.v20221123-1800 (org.eclipse.jdt.feature.group 3.18.1400.v20221123-1800)
    To: org.eclipse.equinox.p2.iu; org.eclipse.jdt.core [3.32.0.v202212121946-e2212,3.32.0.v202212121946-e2212]
```

**Comment by eric-milles** [comment]
**Only one of the following can be installed at once:** 
    Java Development Tools Core 3.32.0.v202212121946-e2212 (org.eclipse.jdt.core 3.32.0.v202212121946-e2212)
    Java Development Tools Core 3.32.0.v20221108-1853 (org.eclipse.jdt.core 3.32.0.v20221108-1853)
    _Java Development Tools Core 3.31.0.v20220831-1439_ (org.eclipse.jdt.core 3.31.0.v20220831-1439)


Any idea where 3.31.0 is coming from?

**Comment by N1k145** [comment]
No, but this seems to be only there in the first case. 
I think the main issue is the conflict between your patched version and the original jdt version.

**Comment by flozano** [comment]
Yes, it's only in first case.
I assumed something was wrong with my initial eclipse, and therefore discarded it, but clean eclipse installations have similar problem as you can see.

**Comment by eric-milles** [comment]
Here are the patch artifacts.  I don't see anything misaligned.  You should be able to match the Java Development Tools feature name and version and then the manifest tells the individual plugin version.

feature.xml (specifies version of JDT feature to patch): https://github.com/groovy/groovy-eclipse/blob/master/jdt-patch/e426/Feature-org.codehaus.groovy.jdt.patch/feature.xml

MANIFEST.MF (specifies version of JDT Core plugin): https://github.com/groovy/groovy-eclipse/blob/master/jdt-patch/e426/org.eclipse.jdt.core/META-INF/MANIFEST.MF

**Comment by N1k145** [comment]
I stared at the numbers way too long, they match, and I also don't see a difference (except the versions) between 2022-09 and 2022-12.
But I still can't install it with the same conflict as already stated.

**Comment by eric-milles** [comment]
So this could be a bug in the [eclipse platform](https://github.com/eclipse-platform).

**Comment by N1k145** [comment]
Could be, I tried to install eclipse 2212+groovy with an older (22-09) eclipse installer and have the same issue. So it should not be a change in the bundle resolution but rather in one of the UpdateSites

**Comment by escarecrow** [comment]
i'm having this exact problem after eclipse updated to 2022-12.

i also notice that, if i check for updates...it lists 
Eclipse JDT Core patch for Groovy	4.8.0.v202212121946-e2212 
but in the details it says
"Eclipse JDT Core patch for Groovy" is not applicable to the current configuration and will not be installed.

**Comment by gjevardat** [comment]
Same issue here starting with a fresh 2012-12 eclipse install

**Comment by alcerar** [comment]
I have same issue

**Comment by stanio** [comment]
I've compared with previous Eclipse 4.25:

```
> rg -C5 3.31.0.v20220831-1439 features
features\org.eclipse.jdt_3.18.1300.v20220831-1800\feature.xml
70-
71-   <plugin
72-         id="org.eclipse.jdt.core"
73-         download-size="6866"
74-         install-size="16189"
75:         version="3.31.0.v20220831-1439"
76-         unpack="false"/>
77-
78-   <plugin
79-         id="org.eclipse.jdt.core.formatterapp"
80-         download-size="12"
```

In Eclipse 4.26 I'm seeing:

```
> rg -C5 3.32.0.v20221108-1853 features
features\org.eclipse.jdt_3.18.1400.v20221123-1800\feature.xml
70-
71-   <plugin
72-         id="org.eclipse.jdt.core"
73-         download-size="6884"
74-         install-size="16238"
75:         version="3.32.0.v20221108-1853"
76-         unpack="false"/>
77-
78-   <plugin
79-         id="org.eclipse.jdt.core.formatterapp"
80-         download-size="12"

features\org.eclipse.epp.package.common.feature_4.26.0.20221201-1200\feature.xml
27-   <!-- JDT core is needed to make help work. See https://github.com/eclipse-platform/eclipse.platform.ua/issues/18 -->
28-   <plugin
29-         id="org.eclipse.jdt.core"
30-         download-size="6884"
31-         install-size="16238"
32:         version="3.32.0.v20221108-1853"
33-         unpack="true"/>
34-
35-</feature>
```

Maybe the latter is preventing the `org.eclipse.jdt.core` plugin from being replaced.

**Comment by eric-milles** [comment]
Right, if this dependency did not exist in `org.eclipse.epp.package.common.feature` or it specified a version range, the patch could be applied properly.

**Comment by eric-milles** [comment]
I'm not sure what can be done about this.  `org.eclipse.epp.package.common.feature` has a hard dependency.  I cannot reuse version number `3.32.0.v20221108-1853`; OSGi will just stick with the bundle it already has.  And https://github.com/eclipse-platform/eclipse.platform.ua/issues/18 states that they have already moved from `org.eclipse.jdt.core` to `org.eclipse.jdt.core.compiler.batch` for 4.27 and beyond.

You could request a point-fix release of the eclipse package(s).  Or you can install a lesser package than JEE.  I didn't try any others specifically, but installing from Eclipse Platform Runtime Binary seems to work fine.

**Comment by flozano** [comment]
eclipse plain java bundle without JEE also failed for me

**Comment by stanio** [comment]
In light of https://github.com/eclipse-platform/eclipse.platform.ua/issues/18#issuecomment-1363779736, I have worked this around for me by manually editing:

```diff
--- a/p2/org.eclipse.equinox.p2.engine/profileRegistry/epp.package.jee.profile/1669926842808.profile.gz	2022-12-01 20:34:04.000000000 +0200
+++ b/p2/org.eclipse.equinox.p2.engine/profileRegistry/epp.package.jee.profile/1669926842808.profile.gz	2022-12-23 12:23:10.000000000 +0200
@@ -26347,9 +26347,8 @@ <unit id='org.eclipse.epp.package.common.feature.feature.group' version='4.26.0.20221201-1200' singleton='false'>
         <provided namespace='org.eclipse.equinox.p2.iu' name='org.eclipse.epp.package.common.feature.feature.group' version='4.26.0.20221201-1200'/>
         <provided namespace='org.eclipse.equinox.p2.localization' name='df_LT' version='1.0.0'/>
       </provides>
-      <requires size='3'>
+      <requires size='2'>
         <required namespace='org.eclipse.equinox.p2.iu' name='org.eclipse.epp.package.common' range='[4.26.0.20221201-1200,4.26.0.20221201-1200]'/>
-        <required namespace='org.eclipse.equinox.p2.iu' name='org.eclipse.jdt.core' range='[3.32.0.v20221108-1853,3.32.0.v20221108-1853]'/>
         <required namespace='org.eclipse.equinox.p2.iu' name='org.eclipse.epp.package.common.feature.feature.jar' range='[4.26.0.20221201-1200,4.26.0.20221201-1200]'>
           <filter>
             (org.eclipse.update.install.features=true)
```

This is the last `<timestamp>.profile.gz` available in the `p2/org.eclipse.equinox.p2.engine/profileRegistry/epp.package.jee.profile/` directory.  I have edited it before starting a fresh "Eclipse IDE for Enterprise Java and Web Developers" unzip for the first time.  Dunno if editing it after being previously started may require running it once with [`-clean` and/or `-initialize`](https://help.eclipse.org/latest/topic/org.eclipse.platform.doc.user/tasks/running_eclipse.htm?cp=0_3_0) option.  I've been able to install Groovy-Eclipse 4.8.0 successfully after this manual hack.

FWIW, I haven't edited `features/org.eclipse.epp.package.common.feature_4.26.0.20221201-1200/feature.xml` to drop the `org.eclipse.jdt.core` plugin requirement declaration.

**Comment by jochenberger** [comment]
That workaround also works in existing installation without using `-clean` or `-initialize`.
![grafik](https://user-images.githubusercontent.com/370733/210515747-e6190980-90c7-419d-bd7e-51f3889c6718.png)
Thanks @stanio

**Comment by twwwt** [comment]
I'm having the same problem. Not sure if I understand all of it correctly. So, this issue is caused by a problem in the Eclipse Platform (https://github.com/eclipse-platform/eclipse.platform.ua/issues/18#issuecomment-1363779736) and if this problem gets resolved this issue resolves itself as a result. Is this correct?

**Comment by N1k145** [comment]
@twwwt from my understanding, they fixed the epp package, but will not create a new release for 2022-12. So for this release, you need the workaround described here. For 2023-03 this should be fixed. As 2023-03 M1 and a GDT Snapshot is available, somebody could try to set it up and see if this works.

**Comment by N1k145** [comment]
I just tested the epp java package for 2023-03 M1 with GDT, and it seems to work again.

**Comment by AnamKhairul** [comment]
I am having the same issue. However, I do not see "profileRegistry/epp.package.jee.profile/1669926842808.profile.gz" in the specified folder.

**Comment by stanio** [comment]
>   I do not see "profileRegistry/epp.package.jee.profile/1669926842808.profile.gz" in the specified folder.

Note:

>   This is the last `<timestamp>.profile.gz` available in the `p2/org.eclipse.equinox.p2.engine/profileRegistry/epp.package.jee.profile/` directory.

Look for the file with the biggest `<timestamp>` (numeric) value having a `.profile.gz` suffix.  I've used the Vim editor from Git Bash for Windows distribution, which automatically gunzips the file for editing and gzips it on saving.

**Comment by AnamKhairul** [comment]
Thanks Stanio. The problem is that **profileRegistry/epp.package.jee.profile** does not exist in my installation.

**Comment by stanio** [comment]
Which Eclipse package are you using?  The name of the profile varies, and for example in "Eclipse IDE for Java Developers" the directory is `p2/org.eclipse.equinox.p2.engine/profileRegistry/epp.package.java.profile`.

**Comment by AnamKhairul** [comment]
Eclipse IDE for Enterprise Java and Web Developers

**Comment by stanio** [comment]
No idea what's up with your install.  When I look into `eclipse-jee-2022-12-R-win32-x86_64.zip` I'm seeing `eclipse/p2/org.eclipse.equinox.p2.engine/profileRegistry/epp.package.jee.profile` directory.

**Comment by AnamKhairul** [comment]
Ok, I installed my Eclipse using the installer "eclipse-inst-jre-win64".

**Comment by stanio** [comment]
With the Oomph installer this appears to end up in `$HOME/.p2/org.eclipse.equinox.p2.engine/profileRegistry/<install_path_like>_eclipse.profile`.

**Comment by AnamKhairul** [comment]
Thanks Stanio, I also noticed that but it is not urgent right now.

**Comment by rich7409** [comment]
Hi. Same problem here this morning, on Linux for me. Clean install of `eclipse-jee-2022-12-R-linux-gtk-x86_64.tar.gz` and installing GDT from the update site `https://dist.springsource.org/release/GRECLIPSE/e4.26`:

```
Cannot complete the install because of a conflicting dependency.
  Software being installed: Eclipse Groovy Development Tools 4.8.0.v202212311456-e2212-RELEASE (org.codehaus.groovy.eclipse.feature.feature.group 4.8.0.v202212311456-e2212-RELEASE)
  Software currently installed: Eclipse IDE for Enterprise Java and Web Developers 4.26.0.20221201-1200 (epp.package.jee 4.26.0.20221201-1200)
  Software currently installed: Eclipse Java Development Tools 3.18.1400.v20221123-1800 (org.eclipse.jdt.feature.group 3.18.1400.v20221123-1800)
  Only one of the following can be installed at once: 
    Java Development Tools Core 3.32.0.v20221108-1853 (org.eclipse.jdt.core 3.32.0.v20221108-1853)
    Java Development Tools Core 3.32.0.v202212311456-e2212-RELEASE (org.eclipse.jdt.core 3.32.0.v202212311456-e2212-RELEASE)
  Cannot satisfy dependency:
    From: Eclipse IDE for Enterprise Java and Web Developers 4.26.0.20221201-1200 (epp.package.jee 4.26.0.20221201-1200)
    To: org.eclipse.equinox.p2.iu; org.eclipse.epp.package.common.feature.feature.group [4.26.0.20221201-1200,4.26.0.20221201-1200]
  Cannot satisfy dependency:
    From: Groovy-Eclipse Compilerless feature 4.8.0.v202212311456-e2212-RELEASE (org.codehaus.groovy.compilerless.feature.feature.group 4.8.0.v202212311456-e2212-RELEASE)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.headless.feature.feature.group [4.8.0.v202212311456-e2212-RELEASE,4.8.0.v202212311456-e2212-RELEASE]
  Cannot satisfy dependency:
    From: Eclipse Groovy Development Tools 4.8.0.v202212311456-e2212-RELEASE (org.codehaus.groovy.eclipse.feature.feature.group 4.8.0.v202212311456-e2212-RELEASE)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.compilerless.feature.feature.group [4.8.0.v202212311456-e2212-RELEASE,4.8.0.v202212311456-e2212-RELEASE]
  Cannot satisfy dependency:
    From: Groovy-Eclipse Headless feature 4.8.0.v202212311456-e2212-RELEASE (org.codehaus.groovy.headless.feature.feature.group 4.8.0.v202212311456-e2212-RELEASE)
    To: org.eclipse.equinox.p2.iu; org.codehaus.groovy.jdt.patch.feature.group [4.8.0.v202212311456-e2212-RELEASE,4.8.0.v202212311456-e2212-RELEASE]
  Cannot satisfy dependency:
    From: Eclipse Packaging Project Common Feature 4.26.0.20221201-1200 (org.eclipse.epp.package.common.feature.feature.group 4.26.0.20221201-1200)
    To: org.eclipse.equinox.p2.iu; org.eclipse.jdt.core [3.32.0.v20221108-1853,3.32.0.v20221108-1853]
  Cannot satisfy dependency:
    From Patch: org.codehaus.groovy.jdt.patch.feature.group 4.8.0.v202212311456-e2212-RELEASE Eclipse Java Development Tools 3.18.1400.v20221123-1800 (org.eclipse.jdt.feature.group 3.18.1400.v20221123-1800)
    To: org.eclipse.equinox.p2.iu; org.eclipse.jdt.core [3.32.0.v202212311456-e2212-RELEASE,3.32.0.v202212311456-e2212-RELEASE]
```

**Comment by arysin** [comment]
I've just installed 2023-03 M1 and then https://dist.springsource.org/snapshot/GRECLIPSE/e4.26 installs ok

**Comment by JacquesLeRoux** [comment]
`I've just installed 2023-03 M1 and then https://dist.springsource.org/snapshot/GRECLIPSE/e4.26 installs ok`
I did the same rather using https://groovy.jfrog.io/artifactory/plugins-release/e4.26 but got this issue:
```
Failed to create the part's controls
java.lang.NoClassDefFoundError: org/codehaus/groovy/eclipse/refactoring/formatter/GroovyIndentationService
```

**Comment by JacquesLeRoux** [comment]
After reading https://github.com/groovy/groovy-eclipse/issues/1456 I got back to 2022-09

I initially wanted to fix path issues with JDK 17 following https://s.apache.org/8n6op (using "enabled" rather than "ENABLED", referring to https://s.apache.org/pm5c3) but it did not work

Quite unstable situation :/

**Comment by eric-milles** [comment]
> > I've just installed 2023-03 M1 and then https://dist.springsource.org/snapshot/GRECLIPSE/e4.26 installs ok

> I did the same rather using https://groovy.jfrog.io/artifactory/plugins-release/e4.26 but got this issue

Eclipse 2023-03 M3 is now out (as of Feb 15th).  https://groovy.jfrog.io/artifactory/plugins-snapshot/e4.27 is the correct update site for this and has been patched up to milestone 3 as well.

As always, caution is required when using the bleeding edge Eclipse IDE (non-)releases.

**Comment by JacquesLeRoux** [comment]
I'll wait the official release. Any idea why `org.eclipse.jdt.core.compiler.ignoreUnnamedModuleForSplitPackage=enabled`, as suggested at https://s.apache.org/8n6op, did not work for me? It's very annoying when debugging.

**Comment by eric-milles** [comment]
See #1441.  `xml-apis` as detailed there and in the stackoverflow Q&A as well is providing more than just APIs.  I think you should look to replace it with the API jars that you require above and beyond what is provided in `java.xml` module.  In your current situation, you don't exactly have a split package, but an unintended overlap.  We have found so-called "fat" jars like this in use in our projects and have replaced them with artifacts that are JPMS compliant.  Old Java EE distros like WebSphere and the like provided jars like `xml-apis` that once filled a gap for Java 4 or 5 deployments.  But in a Java 9+ world, those jars should be replaced.

**Comment by JacquesLeRoux** [comment]
Thanks, I'll have a look. BTW is there something like https://www.eclipse.org/downloads/packages/release/2023-03/m2 for m3

**Comment by eric-milles** [comment]
They will probably make them available sometime this week.  Just a guess; you'll need to follow-up with Eclipse.

**Comment by JacquesLeRoux** [comment]
Yes thanks, I already do now :)

**Comment by GourabIX** [comment]
Why is this tagged as `wontfix`? This issue is still there.

**Comment by JacquesLeRoux** [comment]
I guess because it's fixed in 2023-03 (not yet released?)
